src/EditorFeatures/Core/Implementation/Workspaces/EditorTextFactoryService.cs

```diff
@@ -94,7 +94,7 @@ private unsafe SourceText CreateTextFromTemporaryStorage(ISupportDirectMemoryAcc
             using (var reader = new DirectMemoryAccessStreamReader(src + 1, streamLength / sizeof(char) - 1))
             {
                 var buffer = CreateTextBuffer(reader, cancellationToken);
-                return buffer.CurrentSnapshot.AsRoslynText(Encoding.Unicode);
+                return buffer.CurrentSnapshot.AsRoslynText(encoding ?? Encoding.Unicode);
```

Comment:
I didn’t think that encoding put a BOM automatically – I think you have to call GetPreamble to get one.
--
Kevin Pilch-Bisson
    http://twitter.com/Pilchie

From: Jared Parsons [mailto:notifications@github.com]
Sent: Tuesday, February 10, 2015 3:00 PM
To: dotnet/roslyn
Subject: Re: [roslyn] Fix 353 (#384)


In src/EditorFeatures/Core/Implementation/Workspaces/EditorTextFactoryService.cs<https://github.com/dotnet/roslyn/pull/384#discussion_r24459738>:

> @@ -94,7 +94,7 @@ private unsafe SourceText CreateTextFromTemporaryStorage(ISupportDirectMemoryAcc

>              using (var reader = new DirectMemoryAccessStreamReader(src + 1, streamLength / sizeof(char) - 1))

>              {

>                  var buffer = CreateTextBuffer(reader, cancellationToken);

> -                return buffer.CurrentSnapshot.AsRoslynText(Encoding.Unicode);

> +                return buffer.CurrentSnapshot.AsRoslynText(encoding ?? Encoding.Unicode);

@pharring<https://github.com/pharring>

What I'm having trouble understanding is the following

encoding ?? Encoding.Unicode

That implies it's possible to get here without an encoding. In that case we replace it with an encoding which will add a BOM to a file if this SourceText is later written to disk. That seems to go against most other scenarios where no BOM would be added.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn/pull/384/files#r24459738>.